sormas-backend/src/main/java/de/symeda/sormas/backend/region/GeoShapeProviderEjb.java

```diff
@@ -266,23 +269,53 @@ private void loadDistrictData(String countryName) {
 					continue;
 				}
 
-				Optional<DistrictReferenceDto> districtResult = districts.stream().filter(r -> {
-					String districtName = r.getCaption().replaceAll("\\W", "").toLowerCase();
-					return districtName.contains(shapeDistrictName)
-						|| shapeDistrictName.contains(districtName)
-						|| GeoShapeHelper.similarity(shapeDistrictName, districtName) > 0.7f;
-				}).reduce((r1, r2) -> {
-					// take the result that best fits
-
-					if (r1.getCaption().replaceAll("\\W", "").toLowerCase().equals(shapeDistrictName))
-						return r1;
-					if (r2.getCaption().replaceAll("\\W", "").toLowerCase().equals(shapeDistrictName))
-						return r2;
-
-					return Double.compare(
-						GeoShapeHelper.similarity(r1.getCaption(), shapeDistrictName),
-						GeoShapeHelper.similarity(r2.getCaption(), shapeDistrictName)) <= 0 ? r1 : r2;
-				});
+				Optional<DistrictReferenceDto> districtResult;
+
+				// Use IDs in germany (could also be used for other countries, if fitting externalIDs are provided. Those can then be mapped to the externalID in SORMAS
+				if (countryName.equals("germany")) {
+					String shapeDistrictId = GeoShapeHelper.sniffShapeAttribute(feature, Arrays.asList("ARS"));
+					if (shapeDistrictId == null) {
+						continue;
+					}
+					districtResult = districts.stream().filter(r -> {
+						String districtExtID = r.getExternalId();
+						if (districtExtID == null) {
+							return false;
+						}
+						return districtExtID.contains(shapeDistrictId)
```

Comment:
I'm not sure about using similarity here. From my understanding we are looking at IDs that should be an exact match. It would not be acceptable to use the shape for a district, when the ID is different. E.g. 142421323 vs. 142421324

Reply:
Hmm yes you're right. The problem is that (of course :P) the shapefiles we use use a slightly different approach. I think the shapefiles provide 5-digit-ID's and the external IDs in SORMAS are usually 7 digits. (something like the first two digits always being 11). I can look into this

Reply:
I've removed the similarity check, and added a custom check for the external ID being "110<shapefile ID>", e.g. `11012345` for the district with shapefile-ID `12345`, as that seems to be common for germany.